server/aws-lsp-codewhisperer/src/language-server/agenticChat/errors.ts

```diff
@@ -71,3 +71,194 @@ export function isRequestAbortedError(error: unknown): boolean {
 
     return false
 }
+
+/**
+ * Base class for file operation errors with customer-facing messages
+ */
+export class FileOperationError extends Error {
+    constructor(
+        message: string,
+        public readonly customerMessage: string
+    ) {
+        super(message)
+        this.name = this.constructor.name
+    }
+}
+
+// Common file system errors
+export class DirectoryNotFoundError extends FileOperationError {
+    constructor(message: string) {
+        super(message, 'The directory does not exist. Please create the directory first.')
+    }
+}
+
+export class PermissionDeniedError extends FileOperationError {
```

Comment:
where did we get the list of errors?

Reply:
I organized all the tool errors we were throwing into this file, and also added a few more common errors from nodejs fs